docs/src/main/asciidoc/redis-reference.adoc

```diff
@@ -3,99 +3,646 @@ This guide is maintained in the main Quarkus repository
+This extension provides imperative and reactive APIs and low-level and high-level (type-safe) clients.
 
-If you don't have a Redis service running locally, you can run Redis in a Docker container:
-[source,shell script]
+
+== Installation
+
+If you want to use this extension, you need to add the `io.quarkus:quarkus-redis` extension first.
+In your `pom.xml` file, add:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
 ----
-docker run --name redis -p 7006:6379 -d redis
+<dependency>
+    <groupId>io.quarkiverse.redis</groupId>
+    <artifactId>quarkus-redis</artifactId>
+</dependency>
 ----
 
-Next, while being in https://github.com/vert-x3/vertx-redis-client[`vertx-redis-client`] root folder execute:
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-redis")
+----
+
+With this dependency, you can then inject Redis clients or _datasource_ (high-level, type-safe API), such as:
+
+[source,java]
+----
+import io.quarkiverse.redis.datasource.api.RedisDataSource;
+
+// ...
+@Inject RedisAPI lowLevelClient;
+@Inject RedisDataSource highLevelApi;
+----
+
+More details about the various APIs offered by the quarkus-redis extension are available in the <<apis>> section.
+
+[[apis]]
+== One extension, multiple APIs
 
-[source,shell script]
+This extension provides multiple ways to interact with Redis:
+
+- the _low-level_ Vert.x client: it's a fully reactive, non-blocking, and asynchronous client.
+More details on https://vertx.io/docs/vertx-redis-client/java/[the Vert.x Redis Client documentation].
+Two APIs are exposed: `io.vertx.redis.client.Redis`, and `io.vertx.redis.client.RedisAPI`.
+You will generally use the latter, except if you need to manage connections yourself.
+- the _low-level_ Mutiny variant of the Vert.x API: Unlike the previous one, it exposes a Mutiny API and provides both reactive and imperative methods (suffixed with `andAwait()`).
+Two APIs are exposed: `io.vertx.mutiny.redis.client.Redis` and `io.vertx.mutiny.redis.client.RedisAPI`.
+You will generally use the latter, except if you need to manage connections yourself.
+- a _high-level_ reactive data source: A type-safe, high-level API to interact with Redis.
+This API is fully reactive and asynchronous.
+It exposes a Mutiny API.
+It exposes the `io.quarkiverse.redis.datasource.api.ReactiveRedisDataSource` interface.
+- a _high-level_ imperative data source: A type-safe, high-level API to interact with Redis.
+It is the imperative variant of the reactive data source.
+It exposes the `io.quarkiverse.redis.datasource.api.RedisDataSource` interface.
+
+To help you select the suitable API for you, here are some recommendations:
+
+* If you are building an imperative (_classic_) Quarkus application integrating with Redis: use `io.quarkiverse.redis.datasource.api.RedisDataSource`.
```

Comment:
```suggestion
* If you are building an imperative (_classic_) Quarkus application integrating with Redis: use `io.quarkus.redis.datasource.api.RedisDataSource`.
```

The package name change to `io.quarkus.redis.datasource.api` has to be applied throughout 
```suggestion
* If you are building an imperative (_classic_) Quarkus application integrating with Redis: use `io.quarkiverse.redis.datasource.api.RedisDataSource`.
```

Reply:
Great catch! I initially planned to contribute the new client to quarkiverse.